flow/project.py

```diff
@@ -934,7 +943,15 @@ def _print_progress(x):
                     err.flush()
                 yield _
 
-        _get_job_status = functools.partial(self.get_job_status, ignore_errors=ignore_errors)
+        cached_status = self.document.get('_status', dict())
```

Comment:
To reduce complexity, I propose the following change:
```
try:
    cached_status = self.document['_status']._as_dict()
except KeyError:
    cached_status = dict()
```

Reply:
NIce. I was trying to find a more idiomatic way to write this but didn't move away from calling `get. This is better.